lib/email_tracking_retriever.py

```diff
@@ -95,14 +94,19 @@ def get_trackings(self) -> Dict[str, Tracking]:
           failed_email_ids.append(email_id)
           tqdm.write(f"Unexpected error fetching tracking from email ID {email_id}: "
                      f"{e.__class__.__name__}: {str(e)}: {util.get_traceback_lines()}")
+
     except Exception as e:
       if not self.args.seen:
         print("Fatal unexpected error parsing emails; marking all as unread.")
         self.back_out_of_all()
       raise Exception("Fatal unexpected fatal error when parsing emails") from e
-    finally:
-      self.driver.quit()
 
+    finally:
+      try:
+        self.driver.quit()
+      except Exception as e:
```

Comment:
bad form to swallow generic exceptions like this, what's the reason for it?

Reply:
So the driver does not get initialized unless we hit log_in_if_necessary for amazon and so for BB we don't and the driver is not initialized so we can't call `.quit()`. Not sure a better way to handle this case unless I create another similar function that does this only for amazon

Reply:
why not just check to see if the driver exists? No matter what, it's bad to catch and swallow generic exceptions because something actually wrong might have happened

Reply:
```
finally:
      if(driver.exists())
        self.driver.quit()
 
```
Like so? Sorry still not so familiar with how to do a lot in python.

Reply:
no, just 

```
finally:
  if self.driver:
    self.driver.quit()
```

Reply:
updated, thanks